### PR TITLE
feat(web): add semantic scoring for suggest-affected-components feedback

### PR DESCRIPTION
**Pull request by kdudka**
## What

Add Jaccard-based semantic scoring for `suggest-affected-components` programmatic feedback so that OSIM's component suggestion acceptance data is properly evaluated, and add a POST endpoint test to verify the static-context flow.

## Why

The `suggest-affected-components` endpoint already existed in the CVE feature registry, but programmatic feedback from OSIM was silently unscored — `calculate_semantic_proximity_score` did not handle this feature, so entries were left with an empty `acceptance_score` that could not be retried.

## How to Test

```bash
make test-web   # all 139 tests pass
make check      # format + lint + type-check clean
```

New tests:
- `TestScoreComponentLists` — 7 unit tests for `_score_component_lists` (exact match, case-insensitive, overlap, disjoint, empty lists)
- `TestCalculateSemanticProximityScore.test_components_*` — 4 integration tests for end-to-end scoring
- `test_suggest_affected_components_post_endpoint` — POST variant with static context

## Related Tickets

Related: https://redhat.atlassian.net/browse/AEGIS-432

## Summary by Sourcery

Add semantic scoring support for suggest-affected-components feedback and verify its end-to-end behavior via new unit and endpoint tests.

New Features:
- Introduce Jaccard-based semantic scoring for suggest-affected-components programmatic feedback and include this feature in the set of semantically scored features.

Enhancements:
- Extend semantic proximity calculation logic to handle component list comparisons alongside existing CWE scoring.

Tests:
- Add unit tests for component list scoring edge cases and overlaps, integration tests for component-based semantic proximity scoring, and a POST endpoint test for the suggest-affected-components static-context flow.

### CODE REVIEW

File: src/aegis_ai_web/tests/test_programmatic_feedback.py [hunk 1]
```diff
@@ -523,6 +531,61 @@ def test_json_object_instead_of_array(self):
...
+
+        score = _score_component_lists(["Kernel"], ["kernel"])
+        assert score == 1.0
+
+    def test_overlapping_lists(self):
+        """Test scoring overlapping component lists."""
+        from aegis_ai_web.src.semantic_scoring import _score_component_lists
+
+        score = _score_component_lists(["kernel", "linux-kernel"], ["kernel"])
+        assert 0.0 < score < 1.0
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_programmatic_feedback.py:556]: That would make the test implementation-dependent.

File: src/aegis_ai_web/tests/test_programmatic_feedback.py [hunk 2]
```diff
@@ -796,6 +859,51 @@ async def test_cvss_alias_with_cvss_vectors(self):
...
+    async def test_components_exact_match(self):
+        """Test component list scoring with exact match."""
+        from aegis_ai_web.src.semantic_scoring import calculate_semantic_proximity_score
+
+        score, explanation = await calculate_semantic_proximity_score(
+            suggested='["kernel"]',
+            submitted='["kernel"]',
+            feature="suggest-affected-components",
+        )
+        assert score == 1.0
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_programmatic_feedback.py:878]: That sounds too complicated for what we need.

File: src/aegis_ai_web/tests/test_programmatic_feedback.py [hunk 3]
```diff
@@ -523,6 +531,61 @@ def test_json_object_instead_of_array(self):
...
+class TestScoreComponentLists:
+    """Tests for _score_component_lists function."""
+
+    def test_identical_lists(self):
+        """Test scoring identical component lists."""
+        from aegis_ai_web.src.semantic_scoring import _score_component_lists
+
+        score = _score_component_lists(["kernel"], ["kernel"])
+        assert score == 1.0
+
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_programmatic_feedback.py:543]: Added in 491ec4717e3950d646782a8ac24a78f12456603c.

**Review by superbuggy — APPROVED** [review]
Looks good
